[1.13] Replace bitbucket with github to fix godep error

nikhita opened the pull request:
Related: https://github.com/kubernetes/kubernetes/pull/72138
1.12 version: https://github.com/kubernetes/kubernetes/pull/79525
This fixes `verify-godeps` for `release-1.13`: https://testgrid.k8s.io/sig-release-1.13-all#verify-1.13

This commit has the following changes:

- Replace `bitbucket.org/ww/goautoneg` with `github.com/munnerz/goautoneg`.
- Replace `bitbucket.org/bertimus9/systemstat` with `github.com/nikhita/systemstat`.
- Bump kube-openapi  so that it's dependency on `bitbucket.org/ww/goautoneg`
moves to `github.com/munnerz/goautoneg`.
- Update `BUILD` files.

---

Bitbucket is replaced with GitHub because:

Atlassian finally pulled the plug on their 1.0 api and forces everyone
to use 2.0 now: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/

This leads to an error like:

```
godep: error downloading dep (bitbucket.org/ww/goautoneg): https://api.bitbucket.org/1.0/repositories/ww/goautoneg: 410 Gone
```

This was fixed in upstream go in https://github.com/golang/tools/commit/13ba8ad772dfbf0f451b5dd0679e9c5605afc05d.

---

To fix this in k/k:

1) We'll need to either bump our vendored version
https://github.com/kubernetes/kubernetes/blob/release-1.13/vendor/golang.org/x/tools/go/vcs/vcs.go#L676.
However, this bump brings in _lots_ of changes.

2) We can entirely remove our dependency on bitbucket.

The second point is better because:

1) godep itself vendors in an older version: https://github.com/tools/godep/blob/master/vendor/golang.org/x/tools/go/vcs/vcs.go#L667.
This means that anyone who installs godep directly, without forking it,
will not be able to use it with Kubernetes if we stick to bitbucket.

2) Bumping `golang/x/tools` requires running `godep restore`, which doesn't
work because that uses the 1.0 api...leading to a catch-22 like situation.

/kind failing-test
/assign @liggitt 
/cc @liggitt @dims @sttts @cblecker 

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

nikhita commented on Godeps/LICENSES:
```diff
@@ -97000,80 +97000,80 @@ SOFTWARE.
+                http://www.apache.org/licenses/
+
+
+TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
```
I can't understand why these changes were generated. Also, apart from the whitespace, I don't undertand why there is such a huge diff when everything remains the same? :thinking:

nikhita commented on Godeps/Godeps.json:
```diff
@@ -2585,6 +2580,10 @@
 			"Rev": "cca7078d478f8520f85629ad7c68962d31ed7682"
 		},
+		{
+			"ImportPath": "github.com/nikhita/systemstat",
```
Happy to move into another location, just adding this here to get started with this commit.

https://github.com/nikhita/systemstat

nikhita commented:
/priority important-soon
/sig release
/area dependency

nikhita commented:
/cc @munnerz

hoegaarden commented on vendor/k8s.io/kube-openapi/pkg/generators/openapi.go:
```diff
@@ -336,6 +336,7 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 		}
 		g.Do("return $.OpenAPIDefinition|raw${\nSchema: spec.Schema{\nSchemaProps: spec.SchemaProps{\n", args)
 		g.generateDescription(t.CommentLines)
+		g.Do("Type: []string{\"object\"},\n", nil)
```
@nikhita  Where does this change come from? Do we need that? It generates a huge diff in the api/openapi-spec/swagger.json I am (lacking a lot of context here) a bit uncomfortable with.

hoegaarden replied:
If I read https://github.com/kubernetes/kubernetes/pull/72138#discussion_r258975988 correctly, this should have not really gotten in with the original commit. Can we remove that maybe?

nikhita replied:
Looking into it, I bumped kube-openapi to the commit in the other PR and these changes are expected with a bump to that commit (see https://github.com/kubernetes/kube-openapi/compare/c59034cc13d587f5ef4e85ca0ade0c1866ae8e1d...ced9eb3070a5f1c548ef46e8dfe2a97c208d9f03). But I'll take a look if we can avoid bringing these in.

nikhita replied:
Done. `pull-bazel-test` is currently failing on all 1.13 cherry-pick PRs. :/

nikhita committed:
Replace bitbucket with github

This commit has the following changes:

- Replace `bitbucket.org/ww/goautoneg` with `github.com/munnerz/goautoneg`.
- Replace `bitbucket.org/bertimus9/systemstat` with `github.com/nikhita/systemstat`.
- Bump kube-openapi to remove so that it's dependency on `bitbucket.org/ww/goautoneg`
moves to `github.com/munnerz/goautoneg`.
- Generate `swagger.json` generated from the above change.
- Update `BUILD` files.

Bitbucket is replaced with GitHub because:

Atlassian finally pulled the plug on their 1.0 api and forces everyone
to use 2.0 now: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/

This leads to an error like:

```
godep: error downloading dep (bitbucket.org/ww/goautoneg): https://api.bitbucket.org/1.0/repositories/ww/goautoneg: 410 Gone
```

This was fixed in upstream go in golang/tools@13ba8ad.

To fix this in k/k:

1) We'll need to either bump our vendored version
https://github.com/kubernetes/kubernetes/blob/release-1.13/vendor/golang.org/x/tools/go/vcs/vcs.go#L676.
However, this bump brings in _lots_ of changes.

2) We can entirely remove our dependency on bitbucket.

The second point is better because:

1) godep itself vendors in an older version: https://github.com/tools/godep/blob/master/vendor/golang.org/x/tools/go/vcs/vcs.go#L667.
This means that anyone who installs godep directly, without forking it,
will not be able to use it with Kubernetes if we stick to bitbucket.

2) Bumping `golang/x/tools` requires running `godep restore`, which doesn't
work because that uses the 1.0 api...leading to a catch-22 like situation.

liggitt commented:
godep changes lgtm

liggitt committed:
Allow unit test to pass on machines without ipv6

liggitt commented:
fix for ipv6 unit test in https://github.com/kubernetes/kubernetes/pull/79537 if you want to pick it in

nikhita commented:
> fix for ipv6 unit test in #79537 if you want to pick it in

Thanks, done!

liggitt commented:
/lgtm
/approve

nikhita commented:
/assign @sttts 
for approval

cblecker commented:
/approve
